src/chat/models/reflection.rs

```diff
@@ -0,0 +1,51 @@
+use crate::{
+    chat::{chat_trait::ChatLlm, errors::ModelError, History},
+    secret_manager::Secrets,
+};
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use serde_json;
+use std::collections::HashMap;
+use std::path::Path;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ReflectionModel {
+    pub name: String
+}
+
+#[async_trait]
+impl ChatLlm for ReflectionModel {
+    fn name(&self) -> &str {
+        &self.name
+    }
+
+    fn context_size(&self) -> usize {
+        250
+    }
+
+    async fn chat(
+        &self,
+        _secrets: Secrets,
+        prompt: String,
+        _system: Option<String>,
+        history: Vec<History>,
+    ) -> Result<String, ModelError> {
+        // Ok(format!("history: {}, prompt: {}", history, prompt))
```

Comment:
Whats the proper way to add UUID and history here, ideally if the response is cached as well?